### ISSUE TITLE
add auto-formatter (`black`, `yapf`, etc) to GH actions

### ISSUE THREAD

**Issue by OwenPriceSkelly** [Garden-AI/garden#33]
_as a collaborator on the garden project and a natural pedant, I dislike reading or writing code which is poorly-formatted, and the worst kind of formatting is the inconsistent kind. To ease the cognitive (and emotional) burden of reading or writing poorly-formatted code, we should add a GH action to format the code automatically before the repository is tainted_

I currently use `black` locally, but would be happy to switch to `yapf` or something else more configurable to accommodate anyone's strongly-held opinions

# Acceptance Criteria
Given proposed changes in a PR, when running GH actions (sometime before the `flake8` check), the changes should be formatted by an auto-formatter such as `black` to be stylistically consistent with the rest of the codebase

**Comment by steve-barnard** [comment]
@OwenPriceSkelly have you all decided on which format to use (looked like it might have been black)? Also would having a pre-commit hook for formatting be an option? I have seen the pre-commit formatting work well, as this led to fewer issues when formatting conflicts occurred if someone pushed and forgot to pull back in the linted/formatted code.

**Comment by blaiszik** [comment]
Let's discuss this on Tuesday at the Gathertown meeting. I've noticed, esp in some other repos that our commits are really large when only a few lines are actually changed. We should just pick one as a team and go with it :)

**Comment by OwenPriceSkelly** [comment]
agreed! especially like the idea of having it as a pre-commit hook. Super open to alternatives, but at this point I think it's probably the path of least resistance to just use `black` if that's tolerable to everyone

**Comment by steve-barnard** [comment]
Sounds good! If there is any way to share any current configs for the formatting that would be amazing; I know that when using flake8 and black there can be conflicting information on things like line length etc. Would love to know if there are any current examples of parameters you plan on using such as line length set to 79 or black's 88 or if we are ignoring any other formatting rules; I'm sure this is not a massive priority but would likely allow for us to get something set up in the actions workflow as well as document the standards for any future contributors.
